src/apps/logging/logging.cpp

```diff
@@ -195,6 +196,29 @@ namespace ccfapp
       };
       // SNIPPET_END: log_record_prefix_cert
 
+      auto log_record_anonymous =
+        [this](Store::Tx& tx, CallerId caller_id, nlohmann::json&& params) {
+          if (caller_id != INVALID_ID)
+          {
+            return make_error(
+              HTTP_STATUS_BAD_REQUEST,
+              "Only anonymous callers can record anonymous messages");
```

Comment:
This seems like an odd restriction?

Reply:
You're right. It doesn't test anything that's already tested at the framework level.